allennlp/data/dataset_readers/wikitables/table.py

```diff
@@ -0,0 +1,95 @@
+        column names.
+    """
+    def __init__(self,
+                 column_neighbors: Dict[str, List[str]],
+                 cell_neighbors: Dict[str, List[str]]):
+        self._column_neighbors = column_neighbors
+        self._cell_neighbors = cell_neighbors
+
+    @classmethod
+    def read_table_from_tsv(cls, table_filename: str) -> 'Table':
+        column_neighbors = defaultdict(list)
+        cell_neighbors = defaultdict(list)
+        columns = []
+        # We assume the first row is column names.
+        for row_index, line in enumerate(open(table_filename)):
+            line = line.rstrip('\n')
+            if row_index == 0:
+                # Following Sempre's convention for naming columns.
+                columns = ["fb:row.row.%s" % cls._normalize_string(x) for x in line.split('\t')]
+            else:
+                # Following Sempre's convention for naming cells.
+                cells = ["fb:cell.%s" % cls._normalize_string(x) for x in line.split('\t')]
+                assert len(columns) == len(cells), ("Invalid format. Row %d has %d columns, but header "
+                                                    "has %d columns" % (row_index, len(cells), len(columns)))
+                for column, cell in zip(columns, cells):
+                    column_neighbors[column].append(cell)
+                    cell_neighbors[cell].append(column)
+        return cls(column_neighbors, cell_neighbors)
+
+    @staticmethod
+    def _normalize_string(string: str) -> str:
+        """
+        These are the transformation rules used to normalize cell in column names in Sempre.
+        See ``edu.stanford.nlp.sempre.tables.StringNormalizationUtils.characterNormalize`` and
+        ``edu.stanford.nlp.sempre.tables.TableTypeSystem.canonicalizeName``.
+        We reproduce those rules here to normlaize and canonicalize cells and columns in the same way
+        so that we can match them against constants in logical forms appropriately.
+        """
+        # Normalization rules from Sempre
+        # \u201A -> ,
+        string = re.sub("‚", ",", string)
+        string = re.sub("„", ",,", string)
+        string = re.sub("[·・]", ".", string)
+        string = re.sub("…", "...", string)
+        string = re.sub("ˆ", "^", string)
+        string = re.sub("˜", "~", string)
+        string = re.sub("‹", "<", string)
+        string = re.sub("›", ">", string)
+        string = re.sub("[‘’´`]", "'", string)
+        string = re.sub("[“”«»]", "\"", string)
+        string = re.sub("[•†‡]", "", string)
+        string = re.sub("[‐‑–—]", "-", string)
+        string = re.sub("[\\u2E00-\\uFFFF]", "", string)
+        string = re.sub("\\s+", " ", string)
+        # Canonicalization rules from Sempre
+        string = re.sub("[^\\w]", "_", string)
+        string = re.sub("_+", "_", string)
+        string = re.sub("_$", "", string)
+        return string.lower()
+
+    def get_cell_neighbors(self,
```

Comment:
What's the `name_mapping` for?  The caller could just as easily do `graph.get_cell_neighbors(name_mapping[cell])`.  That'd be preferable to making this API less clean.  Same with getting column neighbors.

Reply:
I agree. Changed this.